server/src/main/java/org/opensearch/plugins/SecureAuxTransportSettingsProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+import javax.net.ssl.SSLContext;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * A security settings provider for auxiliary transports.
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public interface SecureAuxTransportSettingsProvider {
+    /**
+     * Fetch an SSLContext as managed by pluggable security provider.
+     * @return an instance of SSLContext.
+     */
+    default Optional<SSLContext> buildSSLContext() throws NoSuchAlgorithmException, KeyManagementException {
```

Comment:
To stay consistent with `SecureTransportSettingsProvider` and:
 - to allow provider to tailor the engine to particular server `AuxTransport` instance
 - to allow provider to distinguish between server and client `SSLContext`s

```suggestion
    default Optional<SSLContext> buildSecureAuxServerTransportEngine(Settings settings, AuxTransport transport) throws NoSuchAlgorithmException, KeyManagementException {
```

Reply:
Got it, makes sense to give that info to security provider.
Additionally tweaked name to `getSecureAuxTransportContext` per @cwperks below suggestion that contexts will likely be fetched not built. Dropped the `Server` as well since we may fetch a client context.